### PR TITLE
feat: add Github Actions workflow file, to use GH Actions

### PR DESCRIPTION
**Pull request by jmargaret2**
### Summary
Created a Github Actions workflow YAML file, based on the currently-existing CircleCI file, for us to migrate our CI/CD pipeline over to Github Actions. 

### Details
_Add more context to describe the  changes..._

### Checks
- [x] Tested changes using `actionlint` (https://github.com/rhysd/actionlint/tree/main?tab=readme-ov-file#actionlint) a static checker for Github Actions workflows

```
(base) ➜  gradescope-api git:(feat-create-github-actions-workflow) actionlint -verbose
verbose: Linting all workflow files in repository: .
verbose: Detected project: /Users/margaretjagger/VSCodeProjects/gradescope-api
verbose: Collected 1 YAML files
verbose: Linting .github/workflows/main.yaml
verbose: Using project at /Users/margaretjagger/VSCodeProjects/gradescope-api
verbose: Found 0 parse errors in 0 ms for .github/workflows/main.yaml
verbose: Rule "shellcheck" was disabled: exec: "shellcheck": executable file not found in $PATH
verbose: Found total 0 errors in 1 ms for .github/workflows/main.yaml
```
Also tested the functionality of the workflow file itself, using act (https://github.com/nektos/act), to run the workflow locally.

```
| 🎉 All complete!

[Deploy API and Automate Tests/deploy-and-test]   ✅  Success - Main Install PDM dependencies and initialize project repository
[Deploy API and Automate Tests/deploy-and-test] ⭐ Run Main Activate Virtual Environment
[Deploy API and Automate Tests/deploy-and-test]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/4] user= workdir=
| source /Users/margaretjagger/VSCodeProjects/gradescope-api/.venv/bin/activate
[Deploy API and Automate Tests/deploy-and-test]   ✅  Success - Main Activate Virtual Environment
[Deploy API and Automate Tests/deploy-and-test] ⭐ Run Main Install pip and other dependencies
[Deploy API and Automate Tests/deploy-and-test]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/5] user= workdir=
[Deploy API and Automate Tests/deploy-and-test]   ✅  Success - Main Install pip and other dependencies
[Deploy API and Automate Tests/deploy-and-test] ⭐ Run Main Format code
[Deploy API and Automate Tests/deploy-and-test]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/6] user= workdir=
| reformatted /Users/margaretjagger/VSCodeProjects/gradescope-api/tests/test_string_length.py
| reformatted /Users/margaretjagger/VSCodeProjects/gradescope-api/src/sylveon/__main__.py
| 
| All done! ✨ 🍰 ✨
| 2 files reformatted, 8 files left unchanged.
[Deploy API and Automate Tests/deploy-and-test]   ✅  Success - Main Format code
[Deploy API and Automate Tests/deploy-and-test] ⭐ Run Main Lint code
[Deploy API and Automate Tests/deploy-and-test]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/7] user= workdir=
| Success: no issues found in 10 source files
[Deploy API and Automate Tests/deploy-and-test]   ✅  Success - Main Lint code
[Deploy API and Automate Tests/deploy-and-test] ⭐ Run Main Start project
[Deploy API and Automate Tests/deploy-and-test]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/8] user= workdir=
| Hello World
| Length of string Hello World is 11
[Deploy API and Automate Tests/deploy-and-test]   ✅  Success - Main Start project
[Deploy API and Automate Tests/deploy-and-test] ⭐ Run Main Run Tests
[Deploy API and Automate Tests/deploy-and-test]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/9] user= workdir=
| ============================= test session starts ==============================
| platform linux -- Python 3.12.3, pytest-8.0.0, pluggy-1.4.0
| rootdir: /Users/margaretjagger/VSCodeProjects/gradescope-api
collected 4 items                                                              
| 
| tests/test_basic_addition.py ..                                          [ 50%]
| tests/test_hello_world.py .                                              [ 75%]
| tests/test_string_length.py .                                            [100%]
| 
| ============================== 4 passed in 0.02s ===============================
[Deploy API and Automate Tests/deploy-and-test]   ✅  Success - Main Run Tests
[Deploy API and Automate Tests/deploy-and-test] ⭐ Run Post Setup PDM
[Deploy API and Automate Tests/deploy-and-test]   🐳  docker exec cmd=[node /var/run/act/actions/pdm-project-setup-pdm@v4/dist/cache-save.js] user= workdir=
[Deploy API and Automate Tests/deploy-and-test]   ✅  Success - Post Setup PDM
[Deploy API and Automate Tests/deploy-and-test] ⭐ Run Post Set up Python 3.12
[Deploy API and Automate Tests/deploy-and-test]   🐳  docker exec cmd=[node /var/run/act/actions/actions-setup-python@v5/dist/cache-save/index.js] user= workdir=
[Deploy API and Automate Tests/deploy-and-test]   ✅  Success - Post Set up Python 3.12
[Deploy API and Automate Tests/deploy-and-test] Cleaning up container for job deploy-and-test
[Deploy API and Automate Tests/deploy-and-test] 🏁  Job succeeded
```

- [x] Attached Logs

### Team to Review
_Mention the name of the team to review the PR_

### Reference to the issue
_Mention the issue ID or resources_

### CODE REVIEW

**Comment by KevZhe** [comment]
looks good to me, we can figure out GitHub secrets later to make sure tests pass

**Review by KevZhe — APPROVED** [review]

**Review by SushieeK — APPROVED** [review]
Looks good to me!

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Format code
+        run: pdm format
+
+      # - name: Lint code
+      #   run: pdm lint
+
+      - name: Start project
+        run: pdm run start
+
+      - name: Run Tests
```
Comment (calvinatian) [inline .github/workflows/main.yaml:47]: I'll add a Gradescope test account we can use for CI

**Review by calvinatian — APPROVED** [review]
Added a test account for Gradescope and added the email/password to the repository secrets.

Currently, the account is a `Student` for the `Testing` course: https://www.gradescope.com/courses/753413

In the future we can add another secret for CI requiring instructor accounts.

**Pull request merged**